src/model/class.ts

```diff
@@ -0,0 +1,344 @@
+    this.sourceRange = sourceRange;
+    this.properties = properties;
+    this.methods = methods;
+    this.superClass = superClass;
+    this.mixins = mixins;
+    this.privacy = privacy;
+    this.warnings = warnings;
+    this.abstract = abstract;
+    const summaryTag = jsdocLib.getTag(jsdoc, 'summary');
+    this.summary = (summaryTag && summaryTag.description) || '';
+    this.demos = demos;
+  }
+
+  resolve(document: Document): Feature|undefined {
+    return new Class(this, document);
+  }
+}
+
+declare module '../model/queryable' {
+  interface FeatureKindMap {
+    'class': Class;
+  }
+}
+
+export interface ClassInit {
+  readonly sourceRange: SourceRange|undefined;
+  readonly astNode: any;
+  readonly warnings?: Warning[];
+  readonly summary: string;
+  // TODO(rictic): we don't need both name and className here.
+  readonly name?: string;
+  readonly className?: string;
+  readonly jsdoc?: jsdocLib.Annotation;
+  readonly description: string;
+  readonly properties?: Property[];
+  readonly methods?: Method[];
+  readonly superClass?: ScannedReference|undefined;
+  readonly mixins?: ScannedReference[];
+  readonly abstract: boolean;
+  readonly privacy: Privacy;
+  readonly demos?: Demo[];
+}
+export class Class implements Feature {
+  readonly kinds = new Set(['class']);
+  readonly identifiers = new Set<string>();
+  readonly sourceRange: SourceRange|undefined;
+  readonly astNode: any;
+  readonly warnings: Warning[];
+  readonly summary: string;
+  readonly name: string|undefined;
+
+  /**
+   * @deprecated use the `name` field instead.
+   */
+  get className() {
+    return this.name;
+  }
+  readonly jsdoc: jsdocLib.Annotation|undefined;
+  description: string;
+  readonly properties: Property[] = [];
+  readonly methods: Method[] = [];
+  readonly superClass: Reference|undefined;
+  /**
+   * Mixins that this class declares with `@mixes`.
+   *
+   * Mixins are applied linearly after the superclass, in order from first
+   * to last. Mixins that compose other mixins will be flattened into a
+   * single list. A mixin can be applied more than once, each time its
+   * members override those before it in the prototype chain.
+   */
+  readonly mixins: Reference[] = [];
+  readonly abstract: boolean;
+  readonly privacy: Privacy;
+  readonly demos: Demo[];
+
+  constructor(init: ClassInit, document: Document) {
+    this.sourceRange = init.sourceRange;
+    this.warnings = init.warnings ? Array.from(init.warnings) : [];
+    if (init.name) {
+      this.identifiers.add(init.name);
+    }
+    this.astNode = init.astNode;
+    this.demos = jsdocLib.extractDemos(init.jsdoc, document.url);
+
+    this.name = init.name || init.className;
+    if (this.name) {
+      this.identifiers.add(this.name);
+    }
+    this.jsdoc = init.jsdoc;
+    this.description = init.description;
+    this.summary = init.summary;
+    this.abstract = init.abstract;
+    this.privacy = init.privacy;
+    this.demos = init.demos || [];
```

Comment:
this is an example where we could use all of the features of destructuring assignment:

```ts
{
  jsdoc: this.jsdoc,
  description: this.description,
  demos: this.demos = [],
}
```